### ISSUE TITLE
About the Evaluation on Pascal Context

### ISSUE THREAD

**Issue by PkuRainBow** [zhanghang1989/PyTorch-Encoding#78]
I noticed that you illustrate a example with Pascal Context dataset,

> python test.py --dataset PContext --model-zoo Encnet_ResNet50_PContext --eval
> # pixAcc: 0.7838, mIoU: 0.4958: 100%|████████████████████████| 1276/1276 [46:31<00:00,  2.19s/it]

And I check your implementation and find that it seems that you compute the mIoU w/o considering the background classes.

Could you give me a guide about how to deal with the background class as our model is trained over 59 classes.

**Comment by zhanghang1989** [comment]
1. if using background as one of the class, You also need to change https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/datasets/pcontext.py#L102 to
``target = np.array(mask).astype('int32')``
2. I agree that padding zero introducing extra noise. I pad -mean/std during the evaluation. It may do better if pad mean/std during the training.

**Comment by PkuRainBow** [comment]
@zhanghang1989 I mean padding zero for the label map~

**Comment by zhanghang1989** [comment]
Actually, it is padding `-1` (ignored during the training), because we use the `_mask_transform()` to handle it.

**Comment by PkuRainBow** [comment]
@zhanghang1989 Thanks, I have got it.

**Comment by PkuRainBow** [comment]
@zhanghang1989 Thanks, I have got it.

But it seems that in your implementation(on ADE20K, Pascal Context) you will compute the mIou w/o background class on the validation set and w background class on test set(by calling the _mask_transform only for validationphase).  Besides, all of your implementations use the background class for training. I am wondering whether should we train over  background classes for both ADE20K and Pascal Context. Is  this a standard setting?

```

    if args.eval:
        testset = get_segmentation_dataset(args.dataset, split='val', mode='testval',
                                           transform=input_transform)
    else:
        testset = get_segmentation_dataset(args.dataset, split='test', mode='test',
                                           transform=input_transform)
```

**Comment by zhanghang1989** [comment]
On the testset, we don't have a ground truth mask. We just submitted to the server

**Comment by karenyun** [comment]
@zhanghang1989, do u think if I fill the padding area with the void label for both images and masks that will make sense? For my dataset has no background class, but I set the labels from 0 to num-1.

**Comment by zhanghang1989** [comment]
`label=-1` will be ignored during the training.

**Comment by tianhaijie** [comment]
Hi, @PkuRainBow ,do u  train over background classes for Pascal Context?
